src/Protocol/ActivityPub/Transmitter.php

```diff
@@ -0,0 +1,910 @@
+				$tags[] = ['type' => 'Hashtag', 'href' => $term['url'], 'name' => '#' . $term['term']];
+			} elseif ($term['type'] == TERM_MENTION) {
+				$contact = Contact::getDetailsByURL($term['url']);
+				if (!empty($contact['addr'])) {
+					$mention = '@' . $contact['addr'];
+				} else {
+					$mention = '@' . $term['url'];
+				}
+
+				$tags[] = ['type' => 'Mention', 'href' => $term['url'], 'name' => $mention];
+			}
+		}
+		return $tags;
+	}
+
+	/**
+	 * @brief Adds attachment data to the JSON document
+	 *
+	 * @param array $item Data of the item that is to be posted
+	 * @param text $type Object type
+	 * @return attachment array
+	 */
+
+	private static function createAttachmentList($item, $type)
+	{
+		$attachments = [];
+
+		$arr = explode('[/attach],', $item['attach']);
+		if (count($arr)) {
+			foreach ($arr as $r) {
+				$matches = false;
+				$cnt = preg_match('|\[attach\]href=\"(.*?)\" length=\"(.*?)\" type=\"(.*?)\" title=\"(.*?)\"|', $r, $matches);
+				if ($cnt) {
+					$attributes = ['type' => 'Document',
+							'mediaType' => $matches[3],
+							'url' => $matches[1],
+							'name' => null];
+
+					if (trim($matches[4]) != '') {
+						$attributes['name'] = trim($matches[4]);
+					}
+
+					$attachments[] = $attributes;
+				}
+			}
+		}
+
+		if ($type != 'Note') {
+			return $attachments;
+		}
+
+		/// @todo Replace this with a function that takes all pictures from the post
+		$siteinfo = BBCode::getAttachedData($item['body']);
+
+		if (!empty($siteinfo['image']) &&
+			(($siteinfo['type'] == 'photo') ||
+			!Config::get('system', 'ostatus_not_attach_preview'))) {
+				$imgdata = Image::getInfoFromURL($siteinfo['image']);
+				if ($imgdata) {
+					$attachments[] = ['type' => 'Document',
+							'mediaType' => $imgdata['mime'],
+							'url' => $siteinfo['image'],
+							'name' => null];
+				}
+		}
+
+		return $attachments;
+	}
+
+	/**
+	 * @brief Fetches the "context" value for a givem item array from the "conversation" table
+	 *
+	 * @param array $item
+	 *
+	 * @return string with context url
+	 */
+	private static function fetchContextURLForItem($item)
+	{
+		$conversation = DBA::selectFirst('conversation', ['conversation-href', 'conversation-uri'], ['item-uri' => $item['parent-uri']]);
+		if (DBA::isResult($conversation) && !empty($conversation['conversation-href'])) {
+			$context_uri = $conversation['conversation-href'];
+		} elseif (DBA::isResult($conversation) && !empty($conversation['conversation-uri'])) {
+			$context_uri = $conversation['conversation-uri'];
+		} else {
+			$context_uri = $item['parent-uri'] . '#context';
+		}
+		return $context_uri;
+	}
+
+	private static function fetchSensitive($item_id)
+	{
+		$condition = ['otype' => TERM_OBJ_POST, 'oid' => $item_id, 'type' => TERM_HASHTAG, 'term' => 'nsfw'];
+		return DBA::exists('term', $condition);
+	}
+
+	/**
+	 * @brief Creates a note/article object array
+	 *
+	 * @param array $item
+	 *
+	 * @return object array
```

Comment:
if the return variable could be both (either object or array) please use @return object|array

Or is the return variable an array of the object ? (but in in this case we should know which object you are talking from)